.github/workflows/ci_release.yml

```diff
@@ -0,0 +1,69 @@
+name: CI and Release
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+    # Inputs the workflow accepts.
+    inputs:
+      version:
+        # Friendly description to be shown in the UI instead of 'name'
+        description: "Semver type of new version (major / minor / patch)"
+        # Input has to be provided for the workflow to run
+        required: true
+        type: choice
+        options:
+          - patch
+          - minor
+          - major
+
+jobs:
+  # Dockerfile Linting
+  hadolint:
+    uses: celestiaorg/.github/.github/workflows/reusable_dockerfile_lint.yml@main # yamllint disable-line rule:line-length
+    with:
+      dockerfile: docker/Dockerfile
+
+  yamllint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: celestiaorg/.github/.github/actions/yamllint@main
+
+  markdown-lint:
+    name: Markdown Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - run: |
+          npm install -g markdownlint-cli@0.32.1
+          markdownlint --config .markdownlint.yaml '**/*.md'
```

Comment:
Should we use Makefile here?

Reply:
The reason I didn't move this to the Makefile was because markdown linting is helpful in non go repos, so I would split this out to a composite action in the `celestiaorg/.github` repo instead.